src/lib/calc.ts

```
@@ -308,32 +308,33 @@ export function calculateTotalAmount(transactions: Transaction[], type: string)
 		.reduce((prev, transaction) => {
 			const periods = calculateNumOccurrences(
 				transaction.date,
-				transaction.end_date || transaction.date,
-				transaction.repeat_unit || 'month',
-				transaction.repeat || 1,
+				transaction.end_date,
+				transaction.repeat_unit,
+				transaction.repeat,
 			)
 
 			return prev + transaction.amount * periods
 		}, 0)
 }
 export function calculateNumOccurrences(
 	start: string,
-	end: string,
-	repeatUnit: string,
-	repeat: number,
+	end?: string | null,
+	repeatUnit?: string | null,
+	repeat?: number | null,
 ): number {
+	if (!end) return 1
 	const startDate = new Date(start)
 	const endDate = new Date(end)
-
+	const interval = repeat ?? 1
 	switch (repeatUnit) {
 		case 'day':
-			return (differenceInDays(endDate, startDate) + 1) / repeat
+			return (differenceInDays(endDate, startDate) + 1) / interval
 		case 'week':
-			return (differenceInWeeks(endDate, startDate) + 1) / repeat
+			return (differenceInWeeks(endDate, startDate) + 1) / interval
 		case 'month':
-			return (differenceInMonths(endDate, startDate) + 1) / repeat
+			return (differenceInMonths(endDate, startDate) + 1) / interval
 		case 'year':
-			return (differenceInYears(endDate, startDate) + 1) / repeat
+			return (differenceInYears(endDate, startDate) + 1) / interval
 		default:
 			return 1
```

We should throw an exception here.